pkg/controllers/jobset_controller.go

```diff
@@ -114,8 +119,26 @@ func (r *JobSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, err
 	}
 
-	// If JobSet is already completed or failed, clean up active child jobs.
+	// If JobSet is already completed or failed, clean up active child jobs and requeue if TTLSecondsAfterFinished is set.
 	if jobSetFinished(&js) {
+		if js.Spec.TTLSecondsAfterFinished != nil {
+			expiredAt, err := r.checkIfTTLExpired(ctx, &js)
```

Comment:
Any reason why this function returns a *time.Time instead of a bool? Seems like we just use it as a boolean flag basically

Reply:
It is being used to add a field to the log below when did the TTL expire, if that info isn't specially important to be logged, we can easily switch to bool

Reply:
Ah ok I missed that, sounds good, the current implementation is fine.